### PR TITLE
Fix yum & dnf shellout if exit with nonzero status

### PR DESCRIPTION
**Pull request by vsingh-msys**
## Description

If shell out of python yum & dnf helper is exited with nonzero status then raise an error with stderr message
Now it showing 

```
------
FATAL: RuntimeError: yum_package[yum-utils] ((chef-apply cookbook)::(chef-apply recipe) line 1) had an error: RuntimeError: Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'yum'
```
Instead of `"undefined method `+' for false:FalseClass"`
## Related Issue

Fixes https://github.com/chef/chef/issues/8965

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: lib/chef/provider/package/yum/python_helper.rb [hunk 1]
```diff
@@ -40,9 +40,16 @@ class PythonHelper
...
+            @yum_command ||= begin
+              cmd = which("platform-python", "python", "python2", "python2.7", extra_path: "/usr/libexec") do |f|
+                res = shell_out("#{f} -c 'import yum'")
+                raise res.stderr if res.exitstatus != 0
+
+                true
+              end
+
+              "#{cmd} #{YUM_HELPER}"
+            end
```
Comment (lamont-granquist) [inline lib/chef/provider/package/yum/python_helper.rb]: So I think what I'd like better here would look more like:

```ruby
@yum_command ||= begin
    cmd = which("platform-python", "python", "python2", "python2.7", extra_path: "/usr/libexec") do |f|
              shell_out("#{f} -c 'import yum'").exitstatus == 0
            end
    raise Chef::Exceptions::Package, "cannot find yum libraries, you may need to use dnf_package" unless cmd
    "#{cmd} #{YUM_HELPER}"
  end
```

The code here in this PR has a bug where it'll only try to the first python it finds and if that doesn't have the yum/dnf libraries it'll throw.  It needs to preserve the behavior of searching through all the python executables that it finds to find the one that has the yum/dnf libs.

If you want to go the extra mile you could Chef::Log.trace log the stderr of all the shell_out's that are executed (I don't think users will generally want that output)

**Comment by lamont-granquist** [comment]
👍 in general, this is the direction we should take to close out that issue.

File: lib/chef/provider/package/yum/python_helper.rb (hunk 1 above)
Comment (vsingh-msys) [reply lib/chef/provider/package/yum/python_helper.rb]: @lamont-granquist Updated the required changes. Thankyou!

**Comment by vsingh-msys** [comment]
Updated error message in dnf_package & yum_package vice versa. let me know if further if any feedback.

**Pull request merged**